datafusion/expr/src/built_in_function.rs

```diff
@@ -429,31 +341,130 @@ impl BuiltinScalarFunction {
+        BuiltinScalarFunction::Pi => &["pi"],
+        BuiltinScalarFunction::Power => &["power", "pow"],
+        BuiltinScalarFunction::Radians => &["radians"],
+        BuiltinScalarFunction::Random => &["random"],
+        BuiltinScalarFunction::Round => &["round"],
+        BuiltinScalarFunction::Signum => &["signum"],
+        BuiltinScalarFunction::Sin => &["sin"],
+        BuiltinScalarFunction::Sinh => &["sinh"],
+        BuiltinScalarFunction::Sqrt => &["sqrt"],
+        BuiltinScalarFunction::Tan => &["tan"],
+        BuiltinScalarFunction::Tanh => &["tanh"],
+        BuiltinScalarFunction::Trunc => &["trunc"],
+
+        // conditional functions
+        BuiltinScalarFunction::Coalesce => &["coalesce"],
+        BuiltinScalarFunction::NullIf => &["nullif"],
+
+        // string functions
+        BuiltinScalarFunction::Ascii => &["ascii"],
+        BuiltinScalarFunction::BitLength => &["bit_length"],
+        BuiltinScalarFunction::Btrim => &["btrim"],
+        BuiltinScalarFunction::CharacterLength => {
+            &["character_length", "char_length", "length"]
         }
+        BuiltinScalarFunction::Concat => &["concat"],
+        BuiltinScalarFunction::ConcatWithSeparator => &["concat_ws"],
+        BuiltinScalarFunction::Chr => &["chr"],
+        BuiltinScalarFunction::InitCap => &["initcap"],
+        BuiltinScalarFunction::Left => &["left"],
+        BuiltinScalarFunction::Lower => &["lower"],
+        BuiltinScalarFunction::Lpad => &["lpad"],
+        BuiltinScalarFunction::Ltrim => &["ltrim"],
+        BuiltinScalarFunction::OctetLength => &["octet_length"],
+        BuiltinScalarFunction::Repeat => &["repeat"],
+        BuiltinScalarFunction::Replace => &["replace"],
+        BuiltinScalarFunction::Reverse => &["reverse"],
+        BuiltinScalarFunction::Right => &["right"],
+        BuiltinScalarFunction::Rpad => &["rpad"],
+        BuiltinScalarFunction::Rtrim => &["rtrim"],
+        BuiltinScalarFunction::SplitPart => &["split_part"],
+        BuiltinScalarFunction::StartsWith => &["starts_with"],
+        BuiltinScalarFunction::Strpos => &["strpos"],
+        BuiltinScalarFunction::Substr => &["substr"],
+        BuiltinScalarFunction::ToHex => &["to_hex"],
+        BuiltinScalarFunction::Translate => &["translate"],
+        BuiltinScalarFunction::Trim => &["trim"],
+        BuiltinScalarFunction::Upper => &["upper"],
+        BuiltinScalarFunction::Uuid => &["uuid"],
+
+        // regex functions
+        BuiltinScalarFunction::RegexpMatch => &["regexp_match"],
+        BuiltinScalarFunction::RegexpReplace => &["regexp_replace"],
+
+        // time/date functions
+        BuiltinScalarFunction::Now => &["now"],
+        BuiltinScalarFunction::CurrentDate => &["current_date"],
+        BuiltinScalarFunction::CurrentTime => &["current_time"],
+        BuiltinScalarFunction::DateBin => &["date_bin"],
+        BuiltinScalarFunction::DateTrunc => &["date_trunc", "datetrunc"],
```

Comment:
I like how this formulation makes it easier to understand what functions have aliases and which do not 👍